examples/Example14/src/PrimaryGeneratorAction.cc

```diff
@@ -36,6 +36,17 @@ void PrimaryGeneratorAction::GeneratePrimaries(G4Event *aEvent)
 {
   // this function is called at the begining of event
   //
+
+  G4ThreeVector oldDirection = fParticleGun->GetParticleMomentumDirection();
+  // randomize direction if requested
+  if(fRndmDirection > 0.) {
+   
+    G4double rdir            = 0.5 * fRndmDirection;
+    G4double dx = oldDirection.x() + (2 * G4UniformRand() - 1.) * rdir;
+    G4double dy = oldDirection.y() + (2 * G4UniformRand() - 1.) * rdir;
+    G4double dz = oldDirection.z();
+    fParticleGun->SetParticleMomentumDirection(G4ThreeVector(dx,dy,dz));
+  }
```

Comment:
This looks quite odd as direction randomization keeping the Z components and randomizing X and Y.  Also the input in double precision as magnitude of the random on top of the old direction is weird. Why not using a standard eta-phi gun as in [example13](https://github.com/apt-sim/AdePT/blob/a50453948843ea19a17a6029899d00fa856cbab3/examples/Example13/example13.cu#L125)?